src/prefect/server/schemas/actions.py

```diff
@@ -267,9 +267,9 @@ def remove_old_fields(cls, values):
         default_factory=list,
         description="A list of schedules for the deployment.",
     )
-    concurrency_limit: Optional[PositiveInteger] = Field(
-        default=None, description="The deployment's concurrency limit."
-    )
+    concurrency_limit: Optional[
+        Union[PositiveInteger, schemas.core.ConcurrencyLimitConfig]
+    ] = Field(default=None, description="The deployment's concurrency limit.")
```

Comment:
I think we should avoid overloading the typing on this field in the REST API. Since the integer value of the concurrency limit is contained in `ConcurrencyLimitConfig`, I think we should instead have one field for the limit value, and one field for concurrency options. This will make the REST API easier to consume and work with, and we can still have a type overload in the Python client.

Reply:
Makes sense to me! The union type is only really useful as a user-facing simplification. @desertaxle

And just to call something out. Chris mentioned a few times wanting the _client_ to create the GCL, I think. However, the API changes we're developing here and in #15375 follow the same pattern as `schedules`, in sending the user-submitted values (an int, plus an optional config object) to the server, where our endpoint or model layer will handle creating, updating, or deleting the related GCL.

I think this is fine and helps consolidate logic that we would otherwise have to duplicate in the client and the UI.

Reply:
Would it be better to get rid of the Union type all together in order to simplify the schemas for users?